packages/aws-cdk-lib/core/lib/custom-resource-provider/cross-region-export-providers/export-writer-provider.ts

```diff
@@ -24,6 +24,43 @@ export interface ExportWriterProps {
   readonly region?: string;
 }
 
+/**
+ * Create our own CustomResourceProvider so that we can add a single policy
+ * with a list of ARNs instead of having to create a separate policy statement per ARN.
+ */
+class CRProvider extends CustomResourceProvider {
+  public static getOrCreateProvider(scope: Construct, uniqueid: string, props: CustomResourceProviderProps): CRProvider {
+    const id = `${uniqueid}CustomResourceProvider`;
+    const stack = Stack.of(scope);
+    const provider = stack.node.tryFindChild(id) as CRProvider
+      ?? new CRProvider(stack, id, props);
+
+    return provider;
+  }
+
+  private readonly resourceArns = new Set<string>();
+  constructor(scope: Construct, id: string, props: CustomResourceProviderProps) {
+    super(scope, id, props);
+    this.addToRolePolicy({
+      Effect: 'Allow',
+      Resource: Lazy.list({ produce: () => Array.from(this.resourceArns) }),
+      Action: [
+        'ssm:DeleteParameters',
```

Comment:
Is this required or can we `https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_PutParameter.html#API_PutParameter_RequestSyntax`?
`overwrite` set to true flips the `Put` call from a create to an update.